system_setup/server/server.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2021 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from subiquity.server.server import SubiquityServer
+from system_setup.models.system_server import SystemSetupModel
+import os
+
+
+class SystemSetupServer(SubiquityServer):
+
+    from system_setup.server import controllers as controllers_mod
+    controllers = [
+        "Reporting",
+        "Error",
+        "Userdata",
+        "Locale",
+        "Identity",
+        "WSLConfiguration1",
```

Comment:
WSLConfiguration2 went missing

Reply:
just out of curiosity, is it necessary to alter the list of available controllers based on the options? it makes testing a bit inconvenient because we have to restart the server in between.

Reply:
> just out of curiosity, is it necessary to alter the list of available controllers based on the options? it makes testing a bit inconvenient because we have to restart the server in between.

It is necessary, since wslconf1 and wslconf2 have different API implementations.

Reply:
I mean, is it necessary to hide the wslconf2 endpoint or could it be visible in dry-runs at least?

Reply:
> I mean, is it necessary to hide the wslconf2 endpoint or could it be visible in dry-runs at least?

It is visible in dry-run actually

Reply:
For (automatic) testing of both `wslconf1` and `wslconf2`, the server needs to be restarted in between. Only either endpoint is visible at a time. `wslconf1` is visible when running without `--reconfigure`, whereas `wslconf2` is visible when running with `--reconfigure`.

If both endpoints were visible at the same time _in dry-run mode_, it would speed up and simplify testing on the UI side. Notice that the level of testing is not more than that if "the communication works and an answer is received with the expected data structure". This ensures that any endpoint/protocol changes, later on, would not go unnoticed.

In practice, if `WslConfiguration2` was listed here and not overridden below, it would no longer be necessary to restart the server in tests:

```diff
diff --git a/system_setup/server/server.py b/system_setup/server/server.py
index 6d6486d2..06d21bac 100644
--- a/system_setup/server/server.py
+++ b/system_setup/server/server.py
@@ -28,10 +28,11 @@ class SystemSetupServer(SubiquityServer):
         "Locale",
         "Identity",
         "WSLConfiguration1",
+        "WSLConfiguration2",
     ]
 
     def __init__(self, opts, block_log_dir):
-        if opts.reconfigure:
+        if opts.reconfigure and not opts.dry_run:
             self.controllers = [
                 "Reporting",
                 "Error",
```
According to a quick test, this could work, but I'm not sure if querying both `wslconf1` and `wslconf2` breaks the server's internal state even if they are sending dummy responses?

Reply:
Interesting. I will try to change the logic for this part.

Reply:
@jpnurmi I have fixed this bug, it should work as intented